src/aks-preview/azext_aks_preview/custom.py

```diff
@@ -0,0 +1,899 @@
+        enable_rbac=False if disable_rbac else True,
+        agent_pool_profiles=[agent_pool_profile],
+        linux_profile=linux_profile,
+        service_principal_profile=service_principal_profile,
+        network_profile=network_profile,
+        addon_profiles=addon_profiles,
+        aad_profile=aad_profile)
+
+    # Due to SPN replication latency, we do a few retries here
+    max_retry = 30
+    retry_exception = Exception(None)
+    for _ in range(0, max_retry):
+        try:
+            return sdk_no_wait(no_wait, client.managed_clusters.create_or_update,
+                               resource_group_name=resource_group_name, resource_name=name, parameters=mc)
+        except CloudError as ex:
+            retry_exception = ex
+            if 'not found in Active Directory tenant' in ex.message:
+                time.sleep(3)
+            else:
+                raise ex
+    raise retry_exception
+
+
+def aks_update(cmd, client, resource_group_name, name, enable_cluster_autoscaler=False,
+               disable_cluster_autoscaler=False,
+               update_cluster_autoscaler=False,
+               min_count=None, max_count=None, no_wait=False):
+    if (all([enable_cluster_autoscaler, disable_cluster_autoscaler]) or
+            all([enable_cluster_autoscaler, update_cluster_autoscaler]) or
+            all([disable_cluster_autoscaler, update_cluster_autoscaler])):
+        raise CLIError('specify either "--enable_cluster_autoscaler" or '
+                       '"--disable_cluster_autoscaler" or '
+                       '"--update_cluster_autoscaler".')
+    if all([not enable_cluster_autoscaler, not disable_cluster_autoscaler, not update_cluster_autoscaler]):
+        raise CLIError('specify either "--enable_cluster_autoscaler" or '
+                       '"--disable_cluster_autoscaler" or '
+                       '"--update_cluster_autoscaler".')
+
+    # TODO: change this approach when we support multiple agent pools.
+    instance = client.managed_clusters.get(resource_group_name, name)
+    node_count = instance.agent_pool_profiles[0].count
+
+    if not min_count or not max_count:
+        if enable_cluster_autoscaler or update_cluster_autoscaler:
+            raise CLIError('Please specifying both min_count and max_count when --enable_cluster_autoscaler or'
+                           '--update_cluster_autoscaler set.')
+    if min_count and max_count:
+        if int(min_count) > int(max_count):
+            raise CLIError('value of min_count should be less than or equal to value of max_count.')
+        if int(node_count) < int(min_count) or int(node_count) > int(max_count):
+            raise CLIError("current node count '{}' is not in the range of min_count and max_count.".format(node_count))
+
+    if enable_cluster_autoscaler:
+        if instance.agent_pool_profiles[0].enable_auto_scaling:
+            raise CLIError('Cluster autoscaler is already enabled for this managed cluster.\n'
```

Comment:
this is more of a philosophical azure cli question, is this really an error? if the user is enabling something that is already enabled, does that warrant an error message or is that more of a 'everything is ok but i wont do anything so no error code'?  anybody else can chime in for opinions? this applies to line 529 too.

Reply:
updates should be idempotent to be consistent with the rest of the CLI, this should not raise an error.

Reply:
can the azure-cli owners chime in?

Reply:
It is a good suggestion. For better user experience, we can just throw warning message and return.